src/refiners/foundationals/latent_diffusion/ella_adapter.py

```diff
@@ -0,0 +1,285 @@
+TELLAAdapter = TypeVar("TELLAAdapter", bound="ELLAAdapter[Any]")
+
+
+class LayerNormNoAffine(nn.LayerNorm, fl.Module):
+    def __init__(
+        self,
+        normalized_shape: int | Iterable[int],
+        eps: float = 1e-5,
+        device: Device | str | None = None,
+        dtype: DType | None = None,
+    ) -> None:
+        super().__init__(normalized_shape, eps=eps, elementwise_affine=False, device=device, dtype=dtype)  # type: ignore
+
+
+class TimestepEncoder(fl.Passthrough):
+    def __init__(
+        self,
+        time_embedding_dim: int,
+        time_channel: int,
+        device: Device | str | None = None,
+        dtype: DType | None = None,
+    ) -> None:
+        super().__init__(
+            fl.UseContext("diffusion", "timestep"),
+            RangeEncoder(time_channel, time_embedding_dim, device=device, dtype=dtype),
+            fl.SetContext("ella", "timestep_embedding"),
+        )
+
+
+class SquaredReLU(fl.ReLU):
+    def __init__(self) -> None:
+        super().__init__()
+
+    def forward(self, x: Tensor) -> Tensor:
+        return super().forward(x).pow(2)
+
+
+class AdaLayerNorm(fl.Chain):
+    def __init__(
+        self,
+        embedding_dim: int,
+        time_embedding_dim: int,
+        device: Device | str | None = None,
+        dtype: DType | None = None,
+    ) -> None:
+        super().__init__(
+            fl.Parallel(
+                LayerNormNoAffine(embedding_dim, eps=1e-6, device=device, dtype=dtype),
+                fl.Chain(
+                    fl.UseContext("ella", "timestep_embedding"),
+                    fl.SiLU(),
+                    fl.Linear(time_embedding_dim, embedding_dim * 2, device=device, dtype=dtype),
+                ),
+            ),
+            fl.Lambda(self._scale_shift_tensors),
+        )
+
+        self._init_parameters()
+
+    def _init_parameters(self) -> None:
+        _linear: fl.Linear = self.ensure_find(fl.Linear)
+        nn.init.zeros_(_linear.weight)
+        nn.init.zeros_(_linear.bias)
+
+    def _scale_shift_tensors(self, x: Tensor, time_embedding: Tensor) -> Tensor:
+        shift, scale = time_embedding.chunk(2, dim=-1)
+        return x * (1 + scale) + shift
+
+
+class ParameterInitialized(fl.Parameter):
+    def __init__(
+        self, *dims: int, requires_grad: bool = True, device: Device | str | None = None, dtype: DType | None = None
+    ) -> None:
+        super().__init__(*dims, requires_grad=requires_grad, device=device, dtype=dtype)
+        nn.init.normal_(self.weight, mean=0, std=dims[1] ** 0.5)
+
+
+class Latents(fl.Chain):
+    def __init__(
+        self,
+        num_latents: int,
+        width: int,
+        device: Device | str | None = None,
+        dtype: DType | None = None,
+    ) -> None:
+        super().__init__(
+            ParameterInitialized(
+                num_latents,
+                width,
+                device=device,
+                dtype=dtype,
+            ),
+        )
```

Comment:
nit: I'm wondering if `ParameterInitialized` and `Latents` could be fused together in a single class. Since `Latents` is simply a chain with a single element: `ParameterInitialized`.

Reply:
I tried to follow the same thing you did for DINOV2 with ClassToken

Reply:
ah right, we do this because this looks prettier when printing the model's repr, see https://github.com/finegrain-ai/refiners/pull/132#discussion_r1423714231